### PR TITLE
Add authentication configurations support

### PR DESCRIPTION
**Pull request by danialmalik**
#### Story Link
https://edlyio.atlassian.net/browse/EDE-498

#### PR Description

Modify openedx-caliper-tracking app to support authentications mechanisms (SASL).

#### Checklist before merging:

- [ ] Squased
- [ ] Reviewd

### CODE REVIEW

**Review by ifahadashraf — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -127,38 +127,95 @@ These files should be located at ``/edx/app/edxapp/`` directory, see the example
     }
 
 2. Add the following keys and their values in the ``lms.env.json`` and ``cms.env.json`` files.
+Please note that all parameters in the `PRODUCER_CONFIG` are unique to the broker instances. You
+can set whatever parameters are required for you.
```
Comment (HamzaIbnFarooq) [inline README.rst]: nit:
```suggestion
can set whatever parameters required for your instance.
```

File: README.rst [hunk 2]
```diff
@@ -127,38 +127,95 @@ These files should be located at ``/edx/app/edxapp/`` directory, see the example
...
+        "PRODUCER_CONFIG": {
+            "bootstrap_servers": [
+                "<List of Kafka Brokers URLs>"
+            ],
+            ...
+        },
+
+        "TOPIC_NAME": "<Kafka Topic Name>",
+
+        "ERROR_REPORT_EMAIL": "<Reporting email address>",
```
Comment (HamzaIbnFarooq) [inline README.rst]: For consistency:
```suggestion
        "ERROR_REPORT_EMAIL": "<Reporting Email Address>",
```

File: openedx_caliper_tracking/tasks.py [hunk 3]
```diff
@@ -115,16 +170,20 @@ def send_system_recovery_email(self):
     """
     Send system recovery report to specified email address.
     """
+    reporting_email = settings.CALIPER_KAFKA_SETTINGS.get('ERROR_REPORT_EMAIL')
```
Comment (HamzaIbnFarooq) [inline openedx_caliper_tracking/tasks.py:173]: Is it an email or an email address?

File: openedx_caliper_tracking/tests/test_caliper_kafka.py [hunk 4]
```diff
@@ -15,6 +15,20 @@
 from openedx_caliper_tracking.tests import TEST_DIR_PATH
 
 
+CALIPER_SETTINGS_FIXTURE = {
```
Comment (HamzaIbnFarooq) [inline openedx_caliper_tracking/tests/test_caliper_kafka.py]:
```suggestion
CALIPER_KAFKA_SETTINGS_FIXTURE = {
```

File: openedx_caliper_tracking/tasks.py (hunk 3 above)
Comment (danialmalik) [reply openedx_caliper_tracking/tasks.py:173]: What's the difference? 
This is the email address I guess like abc@def.com

**Review by HamzaIbnFarooq — APPROVED** [review]

File: openedx_caliper_tracking/tasks.py (hunk 3 above)
Comment (HamzaIbnFarooq) [reply openedx_caliper_tracking/tasks.py:173]: reporting_email_address may be a more appropriate name then??

File: README.rst (hunk 1 above)
Comment (danialmalik) [reply README.rst]: Updated 'you' with 'your instance' but don't think 'are' should be removed.

File: README.rst (hunk 2 above)
Comment (danialmalik) [reply README.rst]: Done

File: openedx_caliper_tracking/tests/test_caliper_kafka.py (hunk 4 above)
Comment (danialmalik) [reply openedx_caliper_tracking/tests/test_caliper_kafka.py]: Done

File: openedx_caliper_tracking/tasks.py (hunk 3 above)
Comment (danialmalik) [reply openedx_caliper_tracking/tasks.py:173]: Don't think it makes a difference.

**Review by imhassantariq — APPROVED** [review]

**Pull request merged**